### ISSUE TITLE
unable to install with DKMS

### ISSUE THREAD

**Issue by Oloompa** [anbox/anbox#1271]
**Please describe your problem:**
I am unable to pass the first step of installation. Trying to add the ppa give me an error.

Typing `sudo add-apt-repository ppa:morphis/anbox-support` i get the following logs :

```
More info: https://launchpad.net/~morphis/+archive/ubuntu/anbox-support
Press [ENTER] to continue or Ctrl-c to cancel adding it.

Hit:1 http://ftp.fr.debian.org/debian buster InRelease
Ign:2 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal InRelease                                          
Hit:3 http://security.debian.org/debian-security buster/updates InRelease                                            
Hit:4 http://ftp.fr.debian.org/debian buster-updates InRelease                                                       
Err:5 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release                                            
  404  Not Found [IP: 91.189.95.83 80]
Hit:6 http://packages.cloud.google.com/apt cloud-sdk-bionic InRelease                                                
Hit:7 http://packages.microsoft.com/repos/vscode stable InRelease                                                    
Hit:8 http://fr.archive.ubuntu.com/ubuntu bionic InRelease                                                           
Hit:9 https://download.docker.com/linux/debian buster InRelease                                                 
Hit:10 https://packages.gitlab.com/runner/gitlab-runner/debian buster InRelease
Reading package lists... Done
E: The repository 'http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

**Comment by parithy** [comment]
you need add the OS name properly. its an ubuntu ppa try disco

**Comment by knightpp** [comment]
Edit .list file
`sudo gedit /etc/apt/sources.list.d/morphis-ubuntu-anbox-support-eoan.list`

Add  '[trusted=yes]' after 'deb'
Example:
```
deb [trusted=yes] http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
# deb-src http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
```

**Comment by Oloompa** [comment]
@knightpp thanks i think you are right.
I unblocked myself few days ago using this process:
https://github.com/anbox/anbox-modules

**Comment by muelli** [comment]
well. it's still an issue, though, that the documentation, i.e. https://docs.anbox.io/userguide/install_kernel_modules.html, mentions wrong instructions.

**Comment by morphis** [comment]
@muelli What is wrong with the instructions?

**Comment by muelli** [comment]
https://docs.anbox.io/userguide/install.html writes:

> The installation of Anbox consists of two steps.
> 
>         Install necessary kernel modules
>         Install the Anbox snap
> 
> 

And then, 

> To install the necessary kernel modules, please read Install Kernel Modules.

On that page, https://docs.anbox.io/userguide/install_kernel_modules.html, it says:

> In order to add the PPA to your Ubuntu system please run the following commands:
> 
> $ sudo add-apt-repository ppa:morphis/anbox-support

And that command, `sudo add-apt-repository ppa:morphis/anbox-support`, does not work, i.e. it gives the error described here in this bug report.

**Comment by dieguezz** [comment]
> Edit .list file
> `sudo gedit /etc/apt/sources.list.d/morphis-ubuntu-anbox-support-eoan.list`
> 
> Insert '[trusted=yes]' after 'deb'
> Example:
> 
> ```
> deb [trusted=yes] http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
> # deb-src http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
> ```
After doing this, i have a new error on `apt update`
```
http://ppa.launchpad.net/morphis/anbox-support/ubuntu/dists/eoan/main/binary-i386/Packages  404  Not Found [IP: 91.189.95.83 80]
```

**Comment by Tetriser** [comment]
I'm having the same problem here.

**Comment by Tetriser** [comment]
> > Edit .list file
> > `sudo gedit /etc/apt/sources.list.d/morphis-ubuntu-anbox-support-eoan.list`
> > Insert '[trusted=yes]' after 'deb'
> > Example:
> > ```
> > deb [trusted=yes] http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
> > # deb-src http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
> > ```
> 
> After doing this, i have a new error on `apt update`
> 
> ```
> http://ppa.launchpad.net/morphis/anbox-support/ubuntu/dists/eoan/main/binary-i386/Packages  404  Not Found [IP: 91.189.95.83 80]
> ```
> 
> And also tried with amd64 but
> 
> ```
> http://ppa.launchpad.net/morphis/anbox-support/ubuntu/dists/eoan/main/binary-amd64/Packages  404  Not Found [IP: 91.189.95.83 80]
> ```

I happen to have the same problem

**Comment by Tetriser** [comment]
It appears that the repository points to folders that no longer exists in the server.
![image](https://user-images.githubusercontent.com/5746458/73629168-40d11a00-468d-11ea-9b3d-356366dafca5.png)

**Comment by PrzemekSkw** [comment]
I have the same problem on Ubuntu 19.10.

**Comment by sobhardwaj** [comment]
sudo add-apt-repository ppa:morphis/anbox-support this command not working in  my linux for ubantu 18.04 anyone know this to fix please do reply ...

thanks

**Comment by aliPMPAINT** [comment]
I have the same issue with Kubuntu 19.10

**Comment by stuartcle** [comment]
Any fix for this issue yet? I am using 19.10 also.

Thank you.

Sincerely.

**Comment by aliPMPAINT** [comment]
> Any fix for this issue yet? I am using 19.10 also.
> 
> Thank you.
> 
> Sincerely.

Em, I was doing random things and I accidently fixed it... lol. I don't know what I did exactly...

**Comment by MMGomaa** [comment]
Any fix for this issue yet? I am using Ubuntu 20.04 LTS

Thank you.

**Comment by addicted2sounds** [comment]
same issue

**Comment by fabien7474** [comment]
same issue

**Comment by reverseon** [comment]
is this not resolved yet? i had the same issue with 20.04 LTS

**Comment by aliPMPAINT** [comment]
I don't know if this will work, but what I did is deleting the whole thing(everything anbox related) I had done and then installing manually from
https://github.com/anbox/anbox-modules
(There is a manual on how to in README)
Then (I have uninstalled anbox before this process) I restarted the computer and then this worked for me:
```
$ ls -1 /dev/{ashmem,binder}
/dev/ashmem
/dev/binder

```
Finally, I installed anbox:
`sudo snap install --devmode --beta anbox`
And it just worked, I hope it'll also work for you guys.

**Comment by databoose** [comment]
Same issue, Ubuntu 20.04 LTS

**Comment by xx-coder-xx** [comment]
It didn't install right when I run it it freezes in Starting... and then shut off.

**Comment by israelfaria** [comment]
Hitting the same issue here.

The solution presented here also does not work:
> Edit .list file
> `sudo gedit /etc/apt/sources.list.d/morphis-ubuntu-anbox-support-eoan.list`
> 
> Insert '[trusted=yes]' after 'deb'
> Example:
> 
> ```
> deb [trusted=yes] http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
> # deb-src http://ppa.launchpad.net/morphis/anbox-support/ubuntu eoan main
> ```

What worked for me was changing "eoan" by "disco".

**Comment by kylerlaird** [comment]
I see that https://github.com/anbox/anbox/issues/1452 says "Ubuntu ships both ashmem and binder with its default kernel since 19.10. Now [No?] need to install the DKMS modules."  On an Ubuntu Focal system, I do not find ashmem or binder under /usr/lib/5.4.0*.  Nor do I find /dev/{ashmem,binder}.

**Comment by Pra-shu** [comment]
first try this:
           ** sudo ls /etc/apt/sources.list.d**
you can see some pakages over there which start with "morphis" then 
           **sudo rm -i /etc/apt/sources.list.d/** +**pakage name**  run for each pakage delete then do not try anbox

**Comment by luqkrzy** [comment]
@Pra-shu Thanks, it worked!

**Comment by NickWilde263** [comment]
im still got no valid OpenPGP data found when run again with @Pra-shu solution

**Comment by Pra-shu** [comment]
> im still got no valid OpenPGP data found when run again with @Pra-shu solution

@NickWilde263 try this [https://askubuntu.com/questions/1054697/gpg-no-valid-openpgp-data-found-in-ubuntu](url)

**Comment by databoose** [comment]
Why is this still a problem? Why is installing the kernel module such a hacky mess?

**Comment by NickWilde263** [comment]
> > im still got no valid OpenPGP data found when run again with @Pra-shu solution
> 
> @NickWilde263 try this [https://askubuntu.com/questions/1054697/gpg-no-valid-openpgp-data-found-in-ubuntu](url)

i fixed it without reason

**Comment by plibither8** [comment]
Still an issue here on Ubuntu 20.04!

I've tried the `[trusted=yes]` solution but that only partially works. I still get a 404 on `amd64 Packages` and so `anbox-modules-dkms` is not founded.

```
Err:42 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal/main amd64 Packages
  404  Not Found [IP: 91.189.95.85 80]
```

@morphis, it would be great if you could high-prioritise this. Thanks! :)

**Comment by NickWilde263** [comment]
> Still an issue here on Ubuntu 20.04!
> 
> I've tried the `[trusted=yes]` solution but that only partially works. I still get a 404 on `amd64 Packages` and so `anbox-modules-dkms` is not founded.
> 
> ```
> Err:42 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal/main amd64 Packages
>   404  Not Found [IP: 91.189.95.85 80]
> ```
> 
> @morphis, it would be great if you could high-prioritise this. Thanks! :)

there no distribution for `focal`

**Comment by Fuseteam** [comment]
> > Still an issue here on Ubuntu 20.04!
> > I've tried the `[trusted=yes]` solution but that only partially works. I still get a 404 on `amd64 Packages` and so `anbox-modules-dkms` is not founded.
> > ```
> > Err:42 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal/main amd64 Packages
> >   404  Not Found [IP: 91.189.95.85 80]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @morphis, it would be great if you could high-prioritise this. Thanks! :)
> 
> there no distribution for `focal`

the ppa is not needed for focal, the modules have been shipping by default for a while now

**Comment by NickWilde263** [comment]
> > > Still an issue here on Ubuntu 20.04!
> > > I've tried the `[trusted=yes]` solution but that only partially works. I still get a 404 on `amd64 Packages` and so `anbox-modules-dkms` is not founded.
> > > ```
> > > Err:42 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal/main amd64 Packages
> > >   404  Not Found [IP: 91.189.95.85 80]
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > @morphis, it would be great if you could high-prioritise this. Thanks! :)
> > 
> > 
> > there no distribution for `focal`
> 
> the ppa is not needed for focal, the modules have been shipping by default for a while now

ok (i dont use Ubuntu actually)

**Comment by Fuseteam** [comment]
@NickWilde263 if you don't use ubuntu how are you using a ppa?

**Comment by Fuseteam** [comment]
@plibither8 in case you didn't get the mention earlier: you do not need to install the modules manually on ubuntu 20.04

**Comment by villandra** [comment]
I'm still having the issue on December 13, 2021.  Totally stupid!  Something is extremely wrong with the developers.

**Comment by Fuseteam** [comment]
the real "stupid" part is how many people still don't understand, when it clearly says that **the PPA is is NO LONGER needed on kernels older than 5 and ubuntu base newer than 19.04 which is more than 3 YEARS old**

**Comment by NickWilde263** [comment]
> @NickWilde263 if you don't use ubuntu how are you using a ppa?

i use Debian 10 at that time and im kinda mashing piece together for debian
